File: problems/026. Remove Duplicates from Sorted Array/solution.py
class Solution:
    # Time Complexity: O(n) || Space complexity: O(1)
    def removeDuplicates(self, nums: [int]) -> int:
        insertionIndex = 1
        for i in range(len(nums)-1):
            if nums[i] != nums[i+1]:
                nums[insertionIndex] = nums[i+1]
                insertionIndex += 1
        return insertionIndex


solution = Solution()
print(solution.removeDuplicates([1, 1, 3, 4, 4, 6, 7, 8]))

File: problems/026. Remove Duplicates from Sorted Array/solution.cpp
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        if (nums.empty())
            return 0;
        int slot = 1;
        for (int i = 0; i < nums.size() - 1; i++)
        {
            if (nums[i] != nums[i + 1])
            {
                nums[slot++] = nums[i + 1];
            }
        }
        return slot;
    }
};